core/src/main/scala/org/apache/spark/TaskEndReason.scala

```diff
@@ -127,6 +146,19 @@ case class ExceptionFailure(
   }
 }
 
+class ThrowableSerializationWrapper(var exception: Throwable) extends Serializable {
```

Comment:
Perhaps add a brief comment on the intention of this class (to avoid errors due to unserializable exceptions), and maybe indicate that exception will be null in this case.